### PR TITLE
Boot delay

### PR DESCRIPTION
**Pull request by rgundi**
Add boot_delay and CONFIG_INIT_STACKS=n tests to improve code coverage for init.c.

### CODE REVIEW

**Comment by rgundi** [comment]
@nashif, the build is failing for quark_d2000_crb. Apparently this is making the ROM region overflow. Not sure how to fix this. Can you please check?

File: tests/kernel/profiling/profiling_api/testcase.yaml [hunk 1]
```diff
@@ -3,3 +3,8 @@ tests:
     arch_exclude: nios2 riscv32
     platform_exclude: em_starterkit
     tags: kernel
+  kernel.profiling.initstack:
+    arch_exclude: nios2 riscv32
+    platform_exclude: em_starterkit
+    extra_configs:
+      - CONFIG_INIT_STACKS=n
```
Comment (andrewboie) [inline tests/kernel/profiling/profiling_api/testcase.yaml]: This option is off by default.
Did you intend to set this to "y"?
How are we testing that the stack was properly filled with 0xaa?
Not seeing this doing much that's useful.
And why are you excluding nios2/riscv32/em_starterkit? That makes no sense, this is supposed to work on all arches, and I know for sure that Nios II supports this.

File: tests/kernel/common/testcase.yaml [hunk 2]
```diff
@@ -2,3 +2,6 @@ tests:
   kernel.common:
     tags: kernel
     min_ram: 16
+  kernel.common.bootdelay:
+    extra_configs:
+      - CONFIG_BOOT_DELAY=10
```
Comment (andrewboie) [inline tests/kernel/common/testcase.yaml]: This is being turned on.
But how do we know it actually works? Tests need to prove that the feature works.

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
Please use a descriptive PR title and do not push the branch to the upstream repo, you should push it to your fork!

**Pull request closed without merging**